components/TextFields/src/Private/MDCTextInputCommonFundament.m

```diff
@@ -0,0 +1,944 @@
+  // placeholderLabel itself somehow.
+
+  [self updatePlaceholderAlpha];
+  [self.textInput setNeedsUpdateConstraints];
+}
+
+- (void)setClearButtonColor:(UIColor *)clearButtonColor {
+  if (![_clearButtonColor isEqual:clearButtonColor]) {
+    _clearButtonColor = clearButtonColor;
+    self.clearButtonImage =
+        [self drawnClearButtonImage:self.clearButtonImage.size color:_clearButtonColor];
+  }
+}
+
+- (void)setClearButtonMode:(UITextFieldViewMode)clearButtonMode {
+  _clearButtonMode = clearButtonMode;
+  [self updateClearButton];
+}
+
+- (void)setEnabled:(BOOL)enabled {
+  _enabled = enabled;
+  self.underline.enabled = enabled;
+}
+
+- (UIFont *)font {
+  return self.textInput.font;
+}
+
+- (void)setFont:(UIFont *)font {
+  [self.textInput setFont:font];
+}
+
+- (void)setHidesPlaceholderOnInput:(BOOL)hidesPlaceholderOnInput {
+  _hidesPlaceholderOnInput = hidesPlaceholderOnInput;
+  [self updatePlaceholderAlpha];
+}
+
+- (BOOL)isEditing {
+  return self.textInput.isEditing;
+}
+
+- (NSString *)placeholder {
+  id placeholderString = self.placeholderLabel.text;
+  if ([placeholderString isKindOfClass:[NSString class]]) {
+    return (NSString *)placeholderString;
+  } else if ([placeholderString isKindOfClass:[NSAttributedString class]]) {
+    return [(NSAttributedString *)placeholderString string];
+  }
+
+  return nil;
+}
+
+- (void)setPlaceholder:(NSString *)placeholder {
+  self.placeholderLabel.text = placeholder;
+  [self updatePlaceholderAlpha];
+  [self.textInput setNeedsLayout];
+}
+
+- (NSString *)text {
+  return self.textInput.text;
+}
+
+- (void)setText:(NSString *)text {
+  [self.textInput setText:text];
+}
+
+- (void)setTextColor:(UIColor *)textColor {
+  if (!textColor) {
+    textColor = MDCTextInputTextColor();
+  }
+
+  if (_textColor != textColor) {
+    _textColor = textColor;
+    [self updateColors];
+  }
+}
+
+- (UIEdgeInsets)textContainerInset {
+  UIEdgeInsets textContainerInset = UIEdgeInsetsZero;
+
+  textContainerInset.top = MDCTextInputFullPadding;
+  textContainerInset.bottom = MDCTextInputFullPadding;
+
+  if ([self.textInput.positioningDelegate respondsToSelector:@selector(textContainerInset:)]) {
+    return [self.textInput.positioningDelegate textContainerInset:textContainerInset];
+  }
+  return textContainerInset;
+}
+
+- (void)setTextInput:(UIView<MDCTextInput> *)textInput {
+  _textInput = textInput;
+
+  [_textInput setNeedsLayout];
+}
+
+#pragma mark - Layout
+
+- (void)updateColors {
+  self.textInput.tintColor = self.cursorColor;
+  self.textInput.textColor = self.textColor;
+}
+
+- (void)updateFontsForDynamicType {
+  if (self.mdc_adjustsFontForContentSizeCategory) {
+    UIFont *textFont = [UIFont mdc_preferredFontForMaterialTextStyle:MDCFontTextStyleBody1];
+    self.textInput.font = textFont;
+    self.leadingUnderlineLabel.font = textFont;
+    self.trailingUnderlineLabel.font = textFont;
+    self.placeholderLabel.font = textFont;
+  }
```

Comment:
Consider calling this method updateFonts, and if mdc_adjustFontForContentSizeCategory == NO, set the fonts back to their default value.